### PR TITLE
[9.0][WIP] product_tax_multicompany_default

### PR DESCRIPTION
**Pull request by carlosdauden**
Set product taxes in all companies

@Tecnativa

### CODE REVIEW

File: product_tax_multicompany_default/README.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=================================
+Product Tax Multi Company Default
+=================================
+
+This modules set default taxes when product create and adds a button to set not
```
Comment (pedrobaeza) [inline product_tax_multicompany_default/README.rst]:
```
This module sets the default company taxes for all the existing companies when a product is created. It also adds a button to set all the taxes from other companies matching them by tax code.
```

File: product_tax_multicompany_default/README.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+default taxes matching by code.
+
```
Comment (pedrobaeza) [inline product_tax_multicompany_default/README.rst]: Add `Configuration` section:

#. As obvious, you need to have several companies.
#. Go to *Accounting > Configuration > Settings*
#. Select the proper company (if you are admin or a user with several companies access), or keep the current one (for a regular user).
#. On *Invoicing & Payments* section, set fields "Default Sale Tax" and "Default Purchase Tax".
```

File: product_tax_multicompany_default/README.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+Usage
+=====
+
+To use this module, you need to:
+
+#. Go to Sales > Sales > Products
```
Comment (pedrobaeza) [inline product_tax_multicompany_default/README.rst]:
```
#. Go to Sales > Sales > Products.
#. Create a new product and save it.
#. Switching user company (or being as admin or user with several companies access), you will see that the product has the default taxes for all the companies.
```

File: product_tax_multicompany_default/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Product Tax Multi Company Default',
+    'summary': """
+        This modules set default taxes when product create and adds
```
Comment (pedrobaeza) [inline product_tax_multicompany_default/__openerp__.py]: Remove this summary or put it correctly (and in a summarized way)

File: product_tax_multicompany_default/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        a button to set not default taxes matching by code.""",
+    'version': '9.0.1.0.0',
+    'category': 'Account',
+    'website': 'http://www.tecnativa.com',
```
Comment (pedrobaeza) [inline product_tax_multicompany_default/__openerp__.py]: https

File: product_tax_multicompany_default/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    'author': 'Tecnativa, '
+              'Odoo Community Association (OCA)',
+    'license': 'AGPL-3',
+    'depends': [
+        'product',
```
Comment (pedrobaeza) [inline product_tax_multicompany_default/__openerp__.py]: This is clearly depending on `account`

File: product_tax_multicompany_default/models/product_template.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+            ]).ids)
+        return taxes_ids
+
+    @api.multi
+    def set_multicompany_taxes(self):
+        self.ensure_one()
+        tax_ids = self.taxes_id.ids
+        supplier_tax_ids = self.supplier_taxes_id.ids
+        company_id = self.env.user.company_id.id
+
```
Comment (pedrobaeza) [inline product_tax_multicompany_default/models/product_template.py]: Don't put empty line breaks

File: product_tax_multicompany_default/models/product_template.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+            taxes_ids.extend(AccountTax.search([
+                ('name', '=', tax.name),
+                ('company_id', '=', company_id)
+            ]).ids)
+        return taxes_ids
+
+    @api.multi
+    def set_multicompany_taxes(self):
+        self.ensure_one()
+        tax_ids = self.taxes_id.ids
```
Comment (pedrobaeza) [inline product_tax_multicompany_default/models/product_template.py]: customer_tax_ids?

File: product_tax_multicompany_default/models/product_template.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        supplier_tax_ids = self.supplier_taxes_id.ids
+        company_id = self.env.user.company_id.id
+
+        self = self.sudo()
+        default_taxes = self.taxes_by_company('taxes_id', company_id)
```
Comment (pedrobaeza) [inline product_tax_multicompany_default/models/product_template.py]: If these are a list of IDs, call it `default_tax_ids`, or better `default_customer_tax_ids`

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by sergiocorato** [comment]
Hi,
I tried this mp, but in a running installation it doesn't work, perhaps I missed something
It work if I change as of https://github.com/Tecnativa/multi-company/pull/1

**Comment by carlosdauden** [comment]
Thanks for your reviews.

@sergiocorato, can you try after fix?

**Comment by carlosdauden** [comment]
I can't reproduce travis errors in local tests.

> 2017-07-25 00:18:38,084 12154 ERROR openerp_test openerp.addons.product_tax_multicompany_default.tests.test_product_tax_multicompany: ` AccessError: ('The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: product.template, Operation: read)', None)


Any problem with test users groups in travis?

**Comment by sergiocorato** [comment]
@carlosdauden: now it works, well done!
:+1:

**Review by pedrobaeza — APPROVED** [review]
Tests fixed. They failed because of module product_multi_company installed alongside. Setting explicitly `company_id=False` in the values you overpass the problem.

**Pull request merged**